### PR TITLE
Update restart files of application tests for p4est 2.3.6

### PR DESCRIPTION
**Pull request by acdaigneault**
### Description

deal.II changed the version of the dependancy *p4est* to 2.3.6 for their docker [master-focal](https://hub.docker.com/layers/dealii/dealii/master-focal/images/sha256-87f2710a593adf8329ffc3ede033204f3273dc44c5b7064e99d9e6284ee44ab6?context=explore) (the one we use for our CI with deal.II version master). We were using *p4est* 2.3.2, which makes some of our application-test restart files not compatible with the new *p4est* version. 
Unfortunately, the previous restart files were generated a long time ago (> 9 months). The .prm has not been updated since then and they were a bit all over the place.

### Solution

Update the .prm files with the proper parameter names.
Update the restart files that were failling. 
Gather the .prm for the lethe-fluid-particles and lethe-fluid-vans .prm generator in the *generator* folder.
If the .prm restart generator file was not there, new ones have been created trying to replicated it.

### Testing

All tests pass on CI with deal.II version master and they are now disabled with deal.II version 9.5.0.

### Documentation

The next PR will modify the installation doc according to *p4est* 2.3.6

### Miscellaneous (will be removed when merged)

Important note: The adaptive sparse contact test with lethe-fluid-particles is now a dummy test since it does not test all the mobility status anymore. I can say if it is a bug, if the .prm files to generated the restart files were wrong or if the small modification in DEM change the status in cfd-dem. I will open an issue about it since the mechanism is not properly tested at the moment see #1182 

I kept all the generated restart files (even .pvdhandler) in case someone wants output from restarts

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Lethe documentation is up to date
- [x] Fix has unit test(s) (preferred) or application test(s)
- [x] The branch is rebased onto master
- [x] Changelog (CHANGELOG.md) is up to date
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent

Pull request related list:
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [x] If the fix is temporary, an issue is opened
- [x] The PR description is cleaned and ready for merge

### CODE REVIEW

File: CHANGELOG.md
Comment (OGaboriault) [inline CHANGELOG.md:10]: ... since the DEM solver has slightly changed since the previous restart files generation and it is now impossible to regenerate the exact same initial condition.

**Review by OGaboriault — APPROVED** [review]
Tiny comment. I think I will work on a script to automatically generate the restart files for every application tests we have.

**Comment by acdaigneault** [comment]
@OGaboriault cool! Also, I did not redo all the restart files, only the ones that were failling and a few more. If we want fully automated generation of everything, we have to regenerate all restart and have a standardization to target them.
I was also thinking about having the restart files generator included in the check parameter script for examples. But the CI job for that one would need a new name

File: .github/workflows/main_release.yml
Comment (OresteMarquis) [inline .github/workflows/main_release.yml:94]: I would add the same else error as the previous comment here.

File: applications_tests/lethe-fluid-particles/CMakeLists.txt
Comment (OresteMarquis) [inline applications_tests/lethe-fluid-particles/CMakeLists.txt:12]: I could not guess why those files were deleted and not the restart_fluidized_bed_files?

File: applications_tests/lethe-fluid-particles/adaptive_sparse_contacts.mpirun=1.output
Comment (OresteMarquis) [inline applications_tests/lethe-fluid-particles/adaptive_sparse_contacts.mpirun=1.output:1]: Does the new p4est version cause the changes in the application_tests? Or did the tests have been changed change?

File: applications_tests/lethe-fluid-particles/generators/adaptative_sparse_contacts_generator_step1.prm
Comment (OresteMarquis) [inline applications_tests/lethe-fluid-particles/generators/adaptative_sparse_contacts_generator_step1.prm:1]: I'm just curious, I don't even know what it does, but why does the output frequency has been set to 0?

File: applications_tests/lethe-fluid/CMakeLists.txt
Comment (OresteMarquis) [inline applications_tests/lethe-fluid/CMakeLists.txt:1]: If I understand correctly, we don't test .vtu anymore?

File: applications_tests/lethe-fluid-vans/dallavalle_packed_bed_files/dem.particles
Comment (OresteMarquis) [inline applications_tests/lethe-fluid-vans/dallavalle_packed_bed_files/dem.particles:1]: If I understand correctly, the .particles, .pvdhandler, .simulationcontrol, etc. are now all in the same file under the pack_bed_files folder? If so, I like it :)

**Review by OresteMarquis — APPROVED** [review]
Not much to say, If all the test work its good for me :) It seems like a lot of reorganization, I like it! Congrats Audrey, It was probably a pain to update everything, thanks for your work :)

File: applications_tests/lethe-fluid-particles/generators/adaptative_sparse_contacts_generator_step1.prm
Comment (blaisb) [reply applications_tests/lethe-fluid-particles/generators/adaptative_sparse_contacts_generator_step1.prm:1]: It means that we do not generate output files. It's not a good idea to generate output files in the tests since i tends to slow down the CI significantly.

File: applications_tests/lethe-fluid-particles/restart_fluidized_bed_files/dem.triangulation.info
Comment (blaisb) [inline applications_tests/lethe-fluid-particles/restart_fluidized_bed_files/dem.triangulation.info:2]: Do you have an idea why this number changed? Anything significant happen or change in the mesh? It's not a big deal.II and also could be a p4est version issue.

File: .github/workflows/main_release.yml
Comment (acdaigneault) [reply .github/workflows/main_release.yml:79]: Is there a documentation of the else statement and error? I don't find it in the GitHub action documentation.
Also, giving an error won't failed the check?

**Review by blaisb — APPROVED** [review]
I'll merge directly after you have addressed @OresteMarquis last suggestion. Very good work. Thank you so much for having taken care of this.

File: applications_tests/lethe-fluid-particles/adaptive_sparse_contacts.mpirun=1.output
Comment (acdaigneault) [reply applications_tests/lethe-fluid-particles/adaptive_sparse_contacts.mpirun=1.output:1]: In this case I'm not sure, but I don't think so. A few things have changed in the DEM solver that causes tiny differences. The mobility status can change in one iteration. 
I build this test and it was pretty hard to get all the status with a very small number of particles. The test is probably very sensitive to the initial condition. Or the generation files in the repo were not the good ones.

File: .github/workflows/main_release.yml
Comment (blaisb) [reply .github/workflows/main_release.yml:79]: If it gives an error it would indeed fail the check.
To keep the PR finalized because it's a critical one, let's postpone this check to a next PR. We can open an issue ot keep track of this idea, but to be honest, right now just writing a comment is all good for the time being. Throwing an error would break the CI and so code: 1 would most likely give an error.
Let's postpone that for the future

**Comment by blaisb** [comment]
@acdaigneault Good to merge on my side, I'll wait for your final GO and then I'll merge

File: applications_tests/lethe-fluid-particles/restart_fluidized_bed_files/dem.triangulation.info
Comment (acdaigneault) [reply applications_tests/lethe-fluid-particles/restart_fluidized_bed_files/dem.triangulation.info:2]: Git does not compare the same file, anyway, as Oreste pointed out, those dem files are not used, I forgot to delete them

File: applications_tests/lethe-fluid-particles/CMakeLists.txt
Comment (acdaigneault) [reply applications_tests/lethe-fluid-particles/CMakeLists.txt:12]: I commited them by mistakes, you have a good eye :)

File: applications_tests/lethe-fluid/CMakeLists.txt
Comment (acdaigneault) [reply applications_tests/lethe-fluid/CMakeLists.txt:1]: I don't think this test was testing that, especially since the test to not write any .vtu. But this is a very good point since no other test seems to test that for lethe-fluid! 
And yeah, I just saw the person that did that was Audrey in 2020, and I guarantee that I did not really know what I was doing back then haha
Do you think it's a good idea to add them back and add outputs to the test?

File: applications_tests/lethe-fluid-vans/dallavalle_packed_bed_files/dem.particles
Comment (acdaigneault) [reply applications_tests/lethe-fluid-vans/dallavalle_packed_bed_files/dem.particles:1]: Yeah, they were all the same!

File: .github/workflows/main_release.yml
Comment (acdaigneault) [reply .github/workflows/main_release.yml:79]: So I don't change anything at the moment?

File: applications_tests/lethe-fluid-particles/generators/adaptative_sparse_contacts_generator_step1.prm
Comment (acdaigneault) [reply applications_tests/lethe-fluid-particles/generators/adaptative_sparse_contacts_generator_step1.prm:1]: Actually, in fluid it does otuput the one at time = 0. Would it be better to write -1 for all tests? In DEM writing 0 does not generate outputs

Comment (blaisb) [reply applications_tests/lethe-fluid-particles/generators/adaptative_sparse_contacts_generator_step1.prm:1]: Let's maybe do that in another PR where we just do that. For the time being let's keep this PR at the curent level it is.

File: .github/workflows/main_release.yml
Comment (blaisb) [reply .github/workflows/main_release.yml:79]: Nope, let's not change anything. Let's merge the PR as-is when you are ready with the current feature set. Then if we want to change other stuff, let's make small PRs for them

**Pull request merged**
